Library/Homebrew/cask/url.rb

```diff
@@ -102,7 +102,7 @@ def call
           result = curl_output("--fail", "--silent", "--location", @uri)
           result.assert_success!
 
-          page = result.stdout
+          page = T.cast(result.stdout, T.all(String, PageWithURL))
           page.extend PageWithURL
```

Comment:
Rarely have I seen a module so clearly named to be a tuple. How much work is it to refactor as a T::Struct consisting of page and url? (I assume the string being decorated is the page?)

Reply:
Something like this, as a starting point: https://github.com/Homebrew/brew/commit/8adce0f8e0dc2e192fcb8fa489f438fb4254fbf5